### PR TITLE
[12.0][ADD] fleet_vehicle_inspection

### PR DESCRIPTION
**Pull request by marcelsavegnago**
#16

### CODE REVIEW

**Comment by marcelsavegnago** [comment]
@brian10048 
image

image

**Comment by mamcode** [comment]
Hello @marcelsavegnago thanks for this work, is this PR ready for review?

**Comment by marcelsavegnago** [comment]
> Hello @marcelsavegnago thanks for this work, is this PR ready for review?

Hello @mamcode thanks.

Yes, I think this PR is ready to review. I was adjusting some tests but it's OK.

File: fleet_vehicle_inspection/views/fleet_vehicle_inspection_item.xml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                </header>
+                <sheet>
+                    <div class="oe_title">
+                        <h1>
+                            <field name="name" readonly="1"/>
+                        </h1>
+                    </div>
+                    <group>
+                        <field name="name"/>
+                    </group>
```
Comment (brian10048) [inline fleet_vehicle_inspection/views/fleet_vehicle_inspection_item.xml]:
```suggestion
                    <div class="oe_title">
                        <label for="name" class="oe_edit_only"/>
                        <h1><field name="name"/></h1>
                    </div>
```

Remove duplicate name field in the view

File: fleet_vehicle_inspection/readme/CONFIGURE.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+No configuration required.
```
Comment (brian10048) [inline fleet_vehicle_inspection/readme/CONFIGURE.rst]:
```suggestion
To configure this module, you will need to set up inspection items for vehicle inspections.

#. Go to Fleet > Configuration > Inspection Items
#. Create or edit inspection items
```

File: fleet_vehicle_inspection/models/fleet_vehicle_inspection.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+                    continue
+                if (inspection.inspection_line_ids.filtered(
+                        lambda x: x.result == 'todo')):
+                    raise UserError(_(
+                        'Inspection cannot be completed. There are uninspected items.'
+                    ))
+                inspection.write({'state': 'confirmed'})
+            else:
+                raise UserError(_(
+                    'Inspection cannot be completed. No there inspected items.'
```
Comment (brian10048) [inline fleet_vehicle_inspection/models/fleet_vehicle_inspection.py]:
```suggestion
                    'Inspection cannot be completed. There are no inspected items.'
```

**Review by brian10048 — COMMENTED** [review]
@marcelsavegnago This has far exceeded the initial expectation and has been done very quickly!  Thank you for such a complete module!

Functionally, everything looks to be working well.

I hate to even ask changes, but I did add a few minor suggestions.

**Review by brian10048 — APPROVED** [review]

**Comment by marcelsavegnago** [comment]
@mamcode can you review please ?

**Comment by marcelsavegnago** [comment]
> @marcelsavegnago This has far exceeded the initial expectation and has been done very quickly! Thank you for such a complete module!

@brian10048 Thanks :). I am prospecting a customer in the car rental industry and the RFC was a good incentive. This module will be useful.

File: fleet_vehicle_inspection/models/fleet_vehicle.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    @api.multi
+    def action_view_inspection(self):
+        action = self.env.ref(
+            "fleet_vehicle_inspection.fleet_vehicle_inspection_act_window").read()[0]
+        if self.inspection_count > 1:
+            action["domain"] = [("id", "in", self.inspection_ids.ids)]
+        else:
+            action['context'] = self.env.context.copy()
+            action['context'].update({'default_vehicle_id': self.id})
```
Comment (mamcode) [inline fleet_vehicle_inspection/models/fleet_vehicle.py]: I think that `default_vehicle_id` should go in context regardless view type of the action. At this time, when I press the button create from the tree view the vehicle is not automatically set.

File: fleet_vehicle_inspection/models/fleet_vehicle_inspection_line.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        readonly=True,
+        copy=False,
+        store=True,
+        default='draft'
+    )
+
+    @api.multi
+    def action_item_success(self):
+        self.write({'result': 'success'})
+        return {}
```
Comment (mamcode) [inline fleet_vehicle_inspection/models/fleet_vehicle_inspection_line.py]:
```suggestion
        return self.write({'result': 'success'})
```

File: fleet_vehicle_inspection/models/fleet_vehicle_inspection_line.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+    @api.multi
+    def action_item_failure(self):
+        self.write({'result': 'failure'})
+        return {}
```
Comment (mamcode) [inline fleet_vehicle_inspection/models/fleet_vehicle_inspection_line.py]:
```suggestion
        return self.write({'result': 'failure'})
```

File: fleet_vehicle_inspection/views/fleet_vehicle_inspection_line.xml [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        <field name="domain">[]</field>
+        <field name="context">{}</field>
+    </record>
+
+<!--    <record model="ir.ui.menu" id="fleet_vehicle_inspection_line_menu">-->
+<!--        <field name="name">Vehicles Inspections Lines</field>-->
+<!--        <field name="parent_id" ref="fleet.fleet_vehicles"/>-->
+<!--        <field name="action" ref="fleet_vehicle_inspection_line_act_window"/>-->
+<!--        <field name="sequence" eval="17"/>-->
+<!--    </record>-->
```
Comment (mamcode) [inline fleet_vehicle_inspection/views/fleet_vehicle_inspection_line.xml]:
```suggestion
```

**Review by mamcode — CHANGES_REQUESTED** [review]
Thanks for the work, only a few small changes...

File: fleet_vehicle_inspection/models/fleet_vehicle.py (hunk 4 above)
Comment (marcelsavegnago) [reply fleet_vehicle_inspection/models/fleet_vehicle.py]: @mamcode try again please. Thanks.

**Review by mamcode — APPROVED** [review]
LGTM, Thanks :+1:

**Comment by mamcode** [comment]
@ivantodorovich Please review this I think that it is ready to merge.

**Comment by ivantodorovich** [comment]
LGTM ! Thanks for the contribution

/ocabot merge nobump

**Pull request merged**